monetio/sat/_omps_nadir_mm.py

```diff
@@ -13,30 +13,36 @@ def read_OMPS_nm(files):
     import xarray as xr
 
     count = 0
-    print(files)
+    #print(files)
     # Check if files are url
     if "https" in files[0]:
         filelist = sorted(files)
         for filename in filelist:
             data = extract_OMPS_nm_opendap(filename)
-            # print(data)
             if count == 0:
                 data_array = data
                 count += 1
             else:
                 data_array = xr.concat([data_array, data], "x")
     else:
-        filelist = sorted(glob(files, recursive=False))
-
+        if isinstance(files,str):
+            filelist = sorted(glob(files, recursive=False))
+        else: 
+            filelist = sorted(files) # assume list
         for filename in filelist:
-            data = extract_OMPS_nm(filename)
-
-            if count == 0:
-                data_array = data
-                count += 1
-            else:
-                data_array = xr.concat([data_array, data], "x")
-
+            try:
+                data = extract_OMPS_nm(filename)
+                if count == 0:
+                    data_array = data
+                    count += 1
+                else:
+                    data_array = xr.concat([data_array, data], "x")
+            except KeyError as e:
+                print(f'KeyError occured for: {filename}')
+                print(e)
+            except ValueError as e:
+                print(f'ValueError occured for: {filename}')
+                print(e)
     return data_array
```

Comment:
```suggestion
    if count == 0:
        raise RuntimeError(f"no files loaded from files={files}")
    return data_array
```
Without something like this, `data_array` would be unbound if all the tries gave `KeyError` or `ValueError` and were skipped (or if `filelist` was empty), which would give a more confusing error message.

Reply:
Thanks for the suggestion. This is likely a more informative place to let the user know there were no files loaded.